includes/controllers/ApiController.php

```diff
@@ -117,4 +121,32 @@ public function getAllPages()
         }
         return new ApiResponse(empty($pagesWithTag) ? null : $pagesWithTag);
     }
+
+    /**
+     * @Route("/api/pages/{id}",options={"acl":{"public"}})
+     */
+    public function getPage($id)
+    {
+        $pageManager = $this->getService(PageManager::class);
+        $diffService = $this->getService(DiffService::class);
+        $entryManager = $this->getService(EntryManager::class);
+        $entryController = $this->getService(EntryController::class);
+        $page = $pageManager->getById($id);
+        $isEntry = $entryManager->isEntry($page['tag']);
+        if ($isEntry)
+            $page['html'] = $entryController->view($page['tag'], $page['time'], false);
+        else
+            $page['html'] = $this->wiki->Format($page["body"], 'wakka', $page['tag']);
+        
+        if (!empty($_GET['includeDiff'])) {
+            $prevVersion = $pageManager->getPreviousRevision($page);
+            $page['commit_diff_html'] = $diffService->getPageDiff($prevVersion, $page, true);
+            if (!$isEntry) $page['commit_diff_code'] = $diffService->getPageDiff($prevVersion, $page, false);
+
+            $lastVersion = $pageManager->getOne($page['tag']);       
+            $page['diff_html'] = $diffService->getPageDiff($page, $lastVersion, true);
+            if (!$isEntry) $page['diff_code'] = $diffService->getPageDiff($page, $lastVersion, false);
+        }
+        return new ApiResponse($page);
```

Comment:
```suggestion
        return new ApiResponse(empty($page) ? null:$page);
```
permettrait d'avoir un formalisme de réponse similaire à toutes les routes api dans le cas d'un retour avec un tableau vide

Reply:
ça renvoit déjà null le pageManage getId, mais j'ai ajusté pour renvoyé un 404 si la page est pas trouvée